otherlibs/stdune/src/fpath.mli

```diff
@@ -29,6 +29,15 @@ val follow_symlinks : string -> string option
 
 val unlink : string -> unit
 val unlink_no_err : string -> unit
+
+type unlink_status =
+  | Success
+  | Is_a_directory
+  | Other_error
+
+(** Unlink and return error, if any. *)
+val unlink_status : string -> unlink_status
```

Comment:
I'm tempted to call this function `unlink` and rename the current `unlink` to `unlink_exn`.

Reply:
I just pushed that too.